DNN Platform/Library/Application/DotNetNukeContext.cs

```diff
@@ -1,105 +1,100 @@
-        {
-            get
-            {
-                if (_current == null)
-                {
-                    _current = new DotNetNukeContext();
-                }
-
-                return _current;
-            }
-
-            set
-            {
-                _current = value;
-            }
-        }
-
-        /// <summary>
-        /// Gets get the application.
-        /// </summary>
-        public Application Application
-        {
-            get
-            {
-                return this._application;
-            }
-        }
-
-        /// <summary>
-        /// Gets the container event listeners. The listeners will be called in each life cycle of load container.
-        /// </summary>
-        /// <see cref="ContainerEventListener"/>
-        /// <seealso cref="DotNetNuke.UI.Containers.Container.OnInit"/>
-        /// <seealso cref="DotNetNuke.UI.Containers.Container.OnLoad"/>
-        /// <seealso cref="DotNetNuke.UI.Containers.Container.OnPreRender"/>
-        /// <seealso cref="DotNetNuke.UI.Containers.Container.OnUnload"/>
-        public IList<ContainerEventListener> ContainerEventListeners
-        {
-            get
-            {
-                return this._containerEventListeners;
-            }
-        }
-
-        /// <summary>
-        /// Gets the skin event listeners. The listeners will be called in each life cycle of load skin.
-        /// </summary>
-        /// <see cref="SkinEventListener"/>
-        /// <seealso cref="DotNetNuke.UI.Skins.Skin.OnInit"/>
-        /// <seealso cref="DotNetNuke.UI.Skins.Skin.OnLoad"/>
-        /// <seealso cref="DotNetNuke.UI.Skins.Skin.OnPreRender"/>
-        /// <seealso cref="DotNetNuke.UI.Skins.Skin.OnUnload"/>
-        public IList<SkinEventListener> SkinEventListeners
-        {
-            get
-            {
-                return this._skinEventListeners;
-            }
-        }
-    }
-}
+    using Microsoft.Extensions.DependencyInjection;
+
+    /// <summary>
+    /// Defines the context for the environment of the DotNetNuke application.
+    /// </summary>
+    public class DotNetNukeContext : IDnnContext
+    {
+        private static IDnnContext current;
+        private readonly IApplicationInfo applicationInfo;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DotNetNukeContext"/> class.
+        /// </summary>
+        /// <remarks>
+        /// Initialize using the public constructor for Dependency Injection, this method will be removed.
+        /// </remarks>
+        [Obsolete("Deprecated in DotNetNuke 9.7.1. This constructor has been replaced by parameterized public constructor which is designed to be used with Dependency Injection. Resolve the new interface 'DotNetNuke.Abstractions.IDnnContext' instead. Scheduled removal in v11.0.0.")]
+        protected DotNetNukeContext()
+            : this(Globals.DependencyProvider.GetRequiredService<IApplicationInfo>())
+        { }

```

Comment:
```suggestion
        { }

        [Obsolete("Deprecated in DotNetNuke 9.7.1. This constructor has been replaced by the overload taking an IApplicationInfo, which should be resolved via Dependency Injection. Scheduled removal in v11.0.0.")]
    protected DotNetNukeContext(Application application)
        : this((IApplicationInfo)application)
    { }
```

Reply:
This makes sense to me, thanks for the help on the verbiage!